src/arrayMethodSort.js

```diff
@@ -1,11 +1,33 @@
+/* eslint-disable no-unused-expressions */
 'use strict';
 
 /**
  * Implement method Sort
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction !== undefined) {
+          compareFunction(this[j], this[i]) < 0
+            ? [this[i], this[j]] = [this[j], this[i]]
+            : [this[i], this[j]];
+        } else if (compareFunction === undefined
+          && typeof this[0] !== 'number') {
+          [this[i], this[j]]
+            = this[j] < this[i]
+              ? [this[i], this[j]] = [this[j], this[i]]
+              : [this[i], this[j]];
+        } else {
+          [this[i], this[j]]
+            = this[j].toString().localeCompare(this[i]) === -1
+              ? [this[j], this[i]]
+              : [this[i], this[j]];
+        }
+      }
```

Comment:
all these checks are based on `is compareFunction undefined or not` 
It won't be if you set default value for argument
create callBack function and pass it as a default value for compareFunction argument

Reply:
then just call compareFunction

Reply:
1 condition that you should check - `swap or not`